packages/apollo-gateway/src/__tests__/integration/value-types.test.ts

```diff
@@ -0,0 +1,57 @@
+import gql from 'graphql-tag';
+import { execute } from '../execution-utils';
+
+import * as accounts from '../__fixtures__/schemas/accounts';
+import * as books from '../__fixtures__/schemas/books';
+import * as inventory from '../__fixtures__/schemas/inventory';
+import * as product from '../__fixtures__/schemas/product';
+import * as reviews from '../__fixtures__/schemas/reviews';
+
+import { astSerializer, queryPlanSerializer } from '../../snapshotSerializers';
+
+expect.addSnapshotSerializer(astSerializer);
+expect.addSnapshotSerializer(queryPlanSerializer);
+
+describe('value types', () => {
```

Comment:
It will take more time, but I think we should verify all types of `ValueTypes` in this test suite instead of just a single object type. We would be remiss if we made a change that broke unions by accident without a strong test coverage.